### PR TITLE
Feature/settings disabled actions

### PR DESCRIPTION
**Pull request by manwithtwowatches**
When query is unsaved we have greyed out options in the settings menu for save as, share, delete and page settings.

### CODE REVIEW

File: public/templates/work_packages.html [hunk 1]
```diff
@@ -63,11 +63,19 @@
...
-      <li ng-show="!query.isNew()"><a href ng-click="showShareModal()"><i class="icon-action-menu icon-publish"></i>{{ I18n.t('js.toolbar.settings.share') }}</a></li>
-      <li ng-show="!query.isNew()"><a href ng-click="showSettingsModal()"><i class="icon-action-menu icon-settings"></i>{{ I18n.t('js.toolbar.settings.page_settings') }}</a></li>
+      <li><a href ng-click="saveQuery()"><i class="icon-action-menu icon-save1"></i>{{ I18n.t('js.toolbar.settings.save') }}</a></li>
+      <li><a href ng-click="showSaveAsModal($event)" ng-class="{'inactive': query.isNew()}">
+        <i class="icon-action-menu icon-save1"></i>{{ I18n.t('js.toolbar.settings.save_as') }}</a>
+      </li>
+      <li><a href ng-click="showShareModal($event)" ng-class="{'inactive': query.isNew()}">
+        <i class="icon-action-menu icon-publish"></i>{{ I18n.t('js.toolbar.settings.share') }}</a>
+      </li>
+      <li><a href ng-click="deleteQuery($event)" ng-class="{'inactive': query.isNew()}">
```
Comment (myabc) [inline public/templates/work_packages.html:73]: This item is styled as inactive, but is still clickable – the delete confirmation dialog will still be displayed. Looks like `preventDisabledAction` should be called from `deleteConfirmed`.

**Comment by marutosi** [comment]
> >    Stupid.
> >    Should not be in description.
> >    All deails should be in ticket and revisions.
> 
> even better! where are your tickets? i've never seen one, have a link?
> anyway, just pasting the summary of your pull request in the description field doesn't hurt, right?!

#1362
Where is ticket link?

**Comment by manwithtwowatches** [comment]
https://www.openproject.org/work_packages/8219
https://www.openproject.org/work_packages/8329

what an amazingly insulting automatically generated complaint that is.

File: public/templates/work_packages.html (hunk 1 above)
Comment (manwithtwowatches) [inline public/templates/work_packages.html:73]: ah yes. well spotted.

**Comment by tilt** [comment]
Is it wanted that "Save" shows the "Save as" dialog when no query is selected and "Save as" itsself is deactivated?

**Comment by manwithtwowatches** [comment]
@tilt Ah that's because it's using the same template. I'll fix this.

**Comment by tilt** [comment]
Well it makes sense the dialog is shown, just wanted to know if it's intentionally triggered by "Save" instead of "Save as"

**Comment by manwithtwowatches** [comment]
Yes it was intentional. Needs a name if it hasn't already been saved.

**Comment by tilt** [comment]
But the deactivated "Save as" is a bit confusing when Save opens the "Save as" dialog isn't it? I'll merge nevertheless and let the testing squad judge this

**Comment by manwithtwowatches** [comment]
I'm pretty sure they will notice.

**Comment by manwithtwowatches** [comment]
Pretty sure Robin "The Machine" Wagner isn't going to let that one slip through.

**Comment by tilt** [comment]
definitely but let's challenge him and test the testers

File: app/assets/javascripts/angular/directives/work_packages/options-dropdown-directive.js [hunk 2]
```diff
@@ -74,38 +74,56 @@ angular.module('openproject.workPackages.directives')
...
+        return true;
+      }
+
       function showModal() {
         scope.$emit('hideAllDropdowns');
         this.activate();
       }
 
+      function showExistingQueryModal(event) {
+        if( preventDisabledAction(event) ){
```
Comment (tilt) [inline app/assets/javascripts/angular/directives/work_packages/options-dropdown-directive.js:121]: btw it's a bit confusing that things are executed when prevent? returns true

Comment (manwithtwowatches) [inline app/assets/javascripts/angular/directives/work_packages/options-dropdown-directive.js:121]: yeah i thought about that and decided that i didn't want to always write !preventDisabledAction(). but you're correct it makes less sense.

Comment (tilt) [inline app/assets/javascripts/angular/directives/work_packages/options-dropdown-directive.js:121]: maybe `actionEnabled`?

Comment (manwithtwowatches) [inline app/assets/javascripts/angular/directives/work_packages/options-dropdown-directive.js:121]: well now i just feel stupid. i deserve an automatically generated insult.

**Comment by myabc** [comment]
> the testing squad

LOL

**Pull request merged**